src/main/java/org/mvc/handlers/MessagesHandler.java

```diff
@@ -0,0 +1,28 @@
+package org.mvc.handlers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Component;
+
+import org.mvc.models.Message;
+import org.mvc.models.IncomingMessage;
+
+@Component
+public class MessagesHandler {
+    private static int nextId = 0;
+    private static ConcurrentHashMap<Integer, Message> storage = new ConcurrentHashMap<>();
+
+    public HttpStatus addSingleMessage(IncomingMessage incomingMessage) {
+        storage.put(nextId, new Message(nextId, incomingMessage.getUser(), incomingMessage.getMessage()));
+        nextId++;
+
+        return HttpStatus.CREATED;
+    }
+
+    public List<Message> getAllUnreadMessages() {
+        return new ArrayList<>(storage.values());
```

Comment:
List.of(storage.values()), is more "idiomatic"

Reply:
Since this project is using `Java8`, method `of` for `List` class is not available yet.
We agreed to leave it as is